### ISSUE TITLE
Fix grid search

### ISSUE THREAD

**Issue by nomomon** [nomomon/text-2-emoji-kitchen#18]
currently the grid search saves the accuracy as the only metric, but it rather should save the `train_loss`, `valid_loss`, `train_acc`, `valid_acc`. Also, would be nice to have the macro-F1 score for train and valid.

Also, selection of the best model based on the validation accuracy is not correct, rather it has to be done by validation loss. (maybe we can ask the TA about this, but afaik validation loss is used)

**Comment by Jazhyc** [comment]
I'll work on this over the weekend
